src/tabular-api-surface.ts

```diff
@@ -507,6 +524,22 @@ Please use the format 'prezzy' or '${instance.name}/tables/prezzy'.`);
         return false;
       };
 
+      requestStream
+        .on(
+          'metadata',
+          (metadata: {internalRepr: Map<string, string[]>; options: {}}) => {
+            metricsCollector?.onMetadataReceived(metadata);
+          }
+        )
+        .on(
+          'status',
+          (status: {
+            metadata: {internalRepr: Map<string, Uint8Array[]>; options: {}};
+          }) => {
+            metricsCollector?.onStatusMetadataReceived(status);
+          }
+        );
```

Comment:
I imagine you'll need to do something like this for every rpc? Should there be a helper for this?

Reply:
Yes. I added `handleStatusAndMetadata` which should help us here.